src/index.tsx

```diff
@@ -23,7 +23,7 @@ class App extends React.Component<AppProps, AppState> {
     if (!d.value) {
       this.setState({ gte: null });
     } else {
-      const dt: DateTime = DateTime.fromJSDate(d.value);
+      const dt: DateTime = DateTime.fromJSDate(d.value).setZone('utc', { keepCalendarTime: true })
```

Comment:
The infinite loop happens because the datepicker sends a date with 00:00 hours, you create a date from it and modify it by setting the zone, which set a different hour, so the library gets a new value, sends a new value and so on... It's quite tricky.

However, in this case, you can solve it by passing the zone as an option to `fromJSDate`, like this:

```suggestion
      const dt = DateTime.fromJSDate(d.value, { zone: "utc" });
```

Apparently, you get the desired behavior even without the `keepCalendarTime` flag.